File: QytetetJava/src/modeloqytetet/Casilla.java
////////////////////////////////////////////////////////////////////////////////
// Autor: Alejandro Manzanares Lemus
// 2º GII A1
////////////////////////////////////////////////////////////////////////////////
package modeloqytetet;

public abstract class Casilla {
    //Atributos
    private int coste = 0; 
    private int numeroCasilla;                                 
    
    //Constructor
    public Casilla(int numeroCasilla) {
        this.numeroCasilla = numeroCasilla;
    }
    
    //Metodos
    public int getCoste() {
        return coste;
    }
    public int getNumeroCasilla() {
        return numeroCasilla;
    }
    protected abstract TipoCasilla getTipo();
    protected abstract TituloPropiedad getTitulo();
    public void setCoste(int coste){
        this.coste = coste;
    }
    protected abstract boolean soyEdificable();
    @Override
    public String toString( ){
        return "Casilla{" + "\n"
            + "numeroCasilla=" + Integer.toString(numeroCasilla) + "\n"
            + "coste=" + Integer.toString(coste) + "\n"
            + "}";
    }
    
    //Fuera del grafico de la P4
//    public Casilla(int numero, TipoCasilla type, TituloPropiedad title){
//        numeroCasilla = numero;
//        tipo = type;
//        setTitulo(title);
//        coste = title.getPrecioCompra();
//    }
//    public Casilla(int numero, TipoCasilla type){
//        numeroCasilla = numero;
//        tipo = type;
//    }     
//    TituloPropiedad asignarPropietario(Jugador jugador){
//        titulo.setPropietario(jugador);
//        return titulo;
//    }
//    int pagarAlquiler(){
//        int costeAlquiler = titulo.pagarAlquiler();
//        return costeAlquiler;
//    }
//    Boolean propietarioEncarcelado(){
//        return titulo.propietarioEncarcelado();
//    }
//    Boolean tengoPropietario(){
//        return titulo.tengoPropietario();
//    }
//    private void setTitulo(TituloPropiedad title) {
//        titulo = title;
//    }
}

File: QytetetRuby/lib/casilla.rb
#///////////////////////////////////////////////////////////////////////////////
#// Autor: Alejandro Manzanares Lemus
#// 2º GII A1
#///////////////////////////////////////////////////////////////////////////////
module ModeloQytetet
  class Casilla
    
    def initialize(numero, type)
      @numero_casilla = numero
      @tipo = type
      @coste = 0
    end
    
    attr_reader :numero_casilla, :tipo, :coste
    
    
    def soy_edificable()
      soy = false
      if @tipo == TipoCasilla::CALLE
        soy = true
      end     
      return soy
    end 
    
    def to_s()
      "NumeroCasilla:#{@numero_casilla}
Tipo:#{@tipo}
Coste:#{@coste}"
    end
    

  end
end
#
#
#def self.crear_casilla_no_titulo(numero, type)
#  self.new(numero, type, nil, 0)
#end
#def self.crear_casilla_titulo(numero, type, title)
#  self.new(numero, type, title, title.precio_compra)
#end
